### PR TITLE
[Config] Fix cannotBeEmpty()

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #24614
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Open for better deprecation message, but this should clarify.

cc @iltar

### CODE REVIEW

**Comment by iltar** [comment]
Small side note, the case where mine triggered, wasn't working with `requiresAtLeastOneElement()`, I've already tried that. I'll check tomorrow what it was used on (ping me in case I forget).

File: src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php [hunk 1]
```diff
@@ -54,12 +54,23 @@ public function providePrototypeNodeSpecificCalls()
...
-            array('cannotBeEmpty', array()),
             array('useAttributeAsKey', array('foo')),
         );
     }
 
     /**
+     * @group legacy
+     * @expectedDeprecation Calling cannotBeEmpty() has no effect, use requiresAtLeastOneElement() instead. In 4.0 both methods will act the same.
+     */
+    public function testPrototypeNodeCannotBeEmpty()
```
Comment (ro0NL) [inline src/Symfony/Component/Config/Tests/Definition/Builder/ArrayNodeDefinitionTest.php]: basically only this + above removal needs to be merged upstream using excpectedException again.

**Comment by ro0NL** [comment]
@iltar from your issue;

>  ->requiresAtLeastOneElement() is not applicable to concrete nodes 

Im not actually sure this relates to the PR of mine :thinking: it should be _->cannotBeEmpty is not..` then.  So curious about the breakage you experiencing :)

**Comment by iltar** [comment]
@ro0NL No, that error is not related to the original issue, but your deprecation is saying `use requiresAtLeastOneElement() instead`, which will also error for me, so it might not be the right description

**Comment by ro0NL** [comment]
Updated deprecation message. @iltar OK for you?

I still find your issue confusing; the title refers to cannotBeEmpty(), yet the body mentions requiresAtLeastOneElement. Im assuming cannotBeEmpty() indeed raises this issue in 3.4

**Comment by iltar** [comment]
Yes, that message sounds good! 👍

**Review by ogizanagi — APPROVED** [review]

**Comment by ogizanagi** [comment]
ping @symfony/deciders

**Review by stof — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Comment by iltar** [comment]
Awesome, thanks for fixing it! :heart:

**Pull request merged**